### PR TITLE
Fixes #14701 - wrong total asset count

### LINKED ISSUE #14701: Wrong total asset count in sidebar
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

After upgrading to v6.4.0 there is a new total asset count in the sidenav. Unfortunately this count is wrong, as it shows the same count as ready to deploy instead of the total count.

### Reproduction steps

1. Open Snipe-IT
2. Look at the sidenav

### Expected behavior

Total asset count instead of ready to deploy.

### Screenshots

snipe-it wrong asset count

### Snipe-IT Version

6.4.0

### Operating System

Debian

### Web Server

Apache

### PHP Version

8.1.25

### Operating System

_No response_

### Browser

_No response_

### Version

_No response_

### Device

_No response_

### Operating System

_No response_

### Browser

_No response_

### Version

_No response_

### Error messages

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Toreg87**
# Description

The total asset count in the sidenav shows the ready to deploy count instead of the total count. Fix this by adjusting the query to all assets. Also respect the setting for archived assets. Add a default value for total assets, since we are now using the settings-variable, which is not available during the setup process.

While at it, move the block for total assets before the ready to deploy assets to match the ordering of the sidenav.

Fixes #14701 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by snipe** [comment]
Thanks!

**Comment by jerm** [comment]
@Toreg87 Thanks again for submitting this! 

In the interest of being helpful, I wanted to call your attention to a follow-up PR to this one: https://github.com/snipe/snipe-it/pull/14711

tl;dr: while your fix worked, your use of all() in `$total_assets = Asset::all()->count();` had the unfortunate effect of slowing down execution and using a lot of memory. The reason is because all() tells the ORM to fetch everything from the database, and then operate on it.  So instead of asking the database to do the count(efficient), php was being asked to do the count, after getting ALL of the data from the database (inefficient).

Leaving out all() makes the query `select count(*) from assets`, vs `select * from assets`. 

In general, use of all() is an anti-pattern and should only be used if absolutely necessary.

Hope this is helpful 🙂   

Thanks again for your contribution!
